### PR TITLE
feat(Dialog): add dialog component

### LINKED ISSUE #371: Create Dialog ui-component
Examples:
image
image
image

### PR DESCRIPTION
**Pull request by zouxuoz**
resolve #371

```js
class SampleModal extends React.Component {

  state = {
    modalIsOpen: false,
  }

  onRequest = ({ close }) => {
    const { router } = this.context;

    return new Promise((resolve) => {
      console.log('send request');
      resolve();
      close(); // close dialog if you want
    });
  }

  openModal = () => {
    this.setState({ modalIsOpen: true });
  }

  closeModal = () => {
    this.setState({ modalIsOpen: false });
  }

  render() {
    const { modalIsOpen } = this.state;

    return (
      <button onClick={this.openModal}>
        open
        <Dialog
          isOpen={modalIsOpen}
          title="title"
          closeText="cancel"
          submitText="submit"
          onClose={this.closeModal}
          onSubmit={this.onRequest}
          danger={true}
        >text</Dialog>
      </button>
    );
  }

}
```

### CODE REVIEW

File: client/components/ui/Dialog/DialogOverlay.css [hunk 1]
```diff
@@ -0,0 +1,14 @@
+@import "styles/colors.css";
+
+.dialog-overlay {
+  align-items: center;
+  background-color: color(black a(80%));
+  display: flex;
+  height: 100%;
```
Comment (rosendi) [inline client/components/ui/Dialog/DialogOverlay.css]: Don't use `width` and `height`, i.e it doesn't guarantee in some cases (I don't remeber why, but I remember the topic). Use:

```
right: 0px;
bottom: 0px;
```

**Comment by zouxuoz** [comment]
```js
class SampleModal extends React.Component {

  state = {
    modalIsOpen: false,
  }

  onRequest = ({ close }) => {
    const { router } = this.context;

    return new Promise((resolve) => {
      console.log('send request');
      resolve();
      close(); // close dialog if you want
    });
  }

  openModal = () => {
    this.setState({ modalIsOpen: true });
  }

  closeModal = () => {
    this.setState({ modalIsOpen: false });
  }

  render() {
    const { modalIsOpen } = this.state;

    return (
      <button onClick={this.openModal}>
        open
        <Dialog
          isOpen={modalIsOpen}
          title="title"
          closeText="cancel"
          submitText="submit"
          onClose={this.closeModal}
          onSubmit={this.onRequest}
          danger={true}
        >text</Dialog>
      </button>
    );
  }

}
```

File: client/components/ui/Dialog/DialogHeader.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React, { PropTypes } from 'react';
+import withBEM from 'containers/withBEM';
+import './DialogHeader.css';
+
+const DialogHeader = ({ block, element, text }) => (
```
Comment (zouxuoz) [inline client/components/ui/Dialog/DialogHeader.js]: fixed

File: client/components/ui/Dialog/DialogOverlay.css (hunk 1 above)
Comment (zouxuoz) [inline client/components/ui/Dialog/DialogOverlay.css]: fixed

**Comment by rosendi** [comment]
lgtm

**Comment by radiansz** [comment]
lgtm

**Pull request merged**